--- serefpolicy-2.5.9/policy/modules/system/udev.fc.udev_tbl	2007-03-20 09:36:50.000000000 -0400
+++ serefpolicy-2.5.9/policy/modules/system/udev.fc	2007-03-22 06:36:55.000000000 -0400
@@ -1,6 +1,6 @@
 # udev
 
-/dev/\.udev(/.*)?	gen_context(system_u:object_r:udev_tbl_t,s0)
+/dev/\.udev(/.*)? --	gen_context(system_u:object_r:udev_tbl_t,s0)
 /dev/\.udevdb	--	gen_context(system_u:object_r:udev_tbl_t,s0)
 /dev/udev\.tbl	--	gen_context(system_u:object_r:udev_tbl_t,s0)
 
