diff --git a/policy/modules/kernel/corecommands.fc b/policy/modules/kernel/corecommands.fc
index 9a8ff3e..593281c 100644
--- a/policy/modules/kernel/corecommands.fc
+++ b/policy/modules/kernel/corecommands.fc
@@ -308,7 +308,6 @@ ifdef(`distro_gentoo',`
 /usr/libexec(/.*)?			gen_context(system_u:object_r:bin_t,s0)
 
 /usr/libexec/git-core/git-shell	--	gen_context(system_u:object_r:shell_exec_t,s0)
-/usr/libexec/cockpit-agent      --  gen_context(system_u:object_r:shell_exec_t,s0)
 /usr/bin/cockpit-bridge         -- gen_context(system_u:object_r:shell_exec_t,s0)
 /usr/libexec/sesh		        --	gen_context(system_u:object_r:shell_exec_t,s0)
 
